src/BrefKernel.php

```diff
@@ -48,11 +48,22 @@ public function getLogDir()
      */
     public function handle($request, $type = HttpKernelInterface::MASTER_REQUEST, $catch = true)
     {
-        if ($this->isLambda() && ! is_dir($this->getCacheDir())) {
-            $this->prepareCacheDir(parent::getCacheDir(), $this->getCacheDir());
-        }
+        $this->prepareCacheDir(parent::getCacheDir(), $this->getCacheDir());
+
+        return parent::handle($request, $type, $catch);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * We also need to prepare the Cache dir in Kernel::boot in case we are in a Console or worker context
+     * in which Kernel::handle is not called.
+     */
+    public function boot()
+    {
+        $this->prepareCacheDir(parent::getCacheDir(), $this->getCacheDir());
 
-        return parent::handle($request, $type, $catch); // TODO: Change the autogenerated stub
+        parent::boot();
```

Comment:
Isn't `boot()` called in the case of HTTP too? Do we risk the case of calling `prepareCacheDir()` twice?

Good idea though, I'm just curious about that specific point.

Reply:
It would be called twice indeed but the second time it would be skipped because of the first `if` statement in `prepareCacheDir`.

`is_dir($writeDir)` would be `true` so it would exit directly, as it does when hitting a warm Lambda.

Reply:
Right, thanks. Then would it make sense to remove the call from handle?

Reply:
Basically `Kernel::boot` was previously used by the bridge but a feature added in Symfony 5.2 introduced `Kernel::preBoot` (obviously executed before `Kernel::boot`) and preBoot was dumping the cache and container before we had a chance to copy it to `/tmp`.

The call in `Kernel::handle` was introduced in https://github.com/brefphp/symfony-bridge/pull/37 to fix this issue with Symfony 5.2 and make the bridge work on all versions of Symfony.

That fixed it for HTTP apps but broke it for Console apps/workers/scheduled commands because `Kernel::handle` is never called in a console context but `kernel::boot` is (and it's ok to put it here because `Kernel::preBoot` is never called either in console).

So I don't see any other way of making it work for both HTTP and Console :slightly_smiling_face:

Reply:
Oh right 😅  thank you for the history, that makes sense then!

Thank you!